src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,13 +3,44 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private V[] values;
+    private K[] keys;
+    private int index;
+
+    public StorageImpl() {
+        values = (V[]) new Object[DEFAULT_CAPACITY];
+        keys = (K[]) new Object[DEFAULT_CAPACITY];
+        index = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
```

Comment:
```suggestion
        if (getIndex(key) >= 0) {
```
same here - let's also save the index into a local variable

Reply:
done!